### PR TITLE
Add release notes for PGD 4.2.1

### PR DESCRIPTION
**Pull request by irionr**
BDR-2785

## What Changed?
Release notes for PGD 4.2.1

### CODE REVIEW

File: product_docs/docs/pgd/4/rel_notes/pgd_4.2.1_rel_notes.mdx [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+| BDR       | 4.2.1   | Bug fix | Don't wait for ADD CONSTRAINT progress if DDL replication is off (BDR-2645, RT86043) <p> The constraint validation from all nodes is not needed if we don't replicate the DDL or from any node that is PARTED or STANDBY. </p>                                                                                           |
+| BDR       | 4.2.1   | Bug fix | Adjust `replication_set_add_ddl_filter` (BDR-2755, RT86825) <p> Replicate all the columns needed for the ddl_filter. </p>                                                                                                                                                                                                |
+| BDR       | 4.2.1   | Bug fix | Add maintanance to bdr.queue catalog table (BDR-2475, RT84105) <p> Remove automatically rows older than the `bdr.max_queue_days` (default disabled) on bdr.queue. </p>                                                                                                                                                   |
```
Comment (mwanner) [inline product_docs/docs/pgd/4/rel_notes/pgd_4.2.1_rel_notes.mdx:18]: s/Remove automatically/Automatically remove/

I would be more explicit with the default and add a separate sentence: "By default, entries are retained forever to match the existing behaviour."

File: product_docs/docs/pgd/4/rel_notes/pgd_4.2.1_rel_notes.mdx [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+| BDR       | 4.2.1   |         | Add timeout option to `bdr_init_physical` (RT8277) <p> The timeout does not take the time spent in other activities of `bdr_init_physical` such as creating a base backup and sending the join request. </p>                                                                                                             |
+| BDR       | 4.2.1   |         | Abort unresolved commit scope transactions on node part (BDR-2597) <p> Add a new peer state PART_TX_RESOLVE. This state will abort any remaining unresolved group commit transactions from the node that's parting on the current raft leader and replicates that to the rest of the cluster. </p>                       |
```
Comment (mwanner) [inline product_docs/docs/pgd/4/rel_notes/pgd_4.2.1_rel_notes.mdx:20]: Let's mark these two as "Bug fix" as well, please.

**Review by mwanner — CHANGES_REQUESTED** [review]

**Review by mwanner — APPROVED** [review]
LGTM

File: product_docs/docs/pgd/4/rel_notes/pgd_4.2.1_rel_notes.mdx [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+| BDR       | 4.2.1   | Bug fix | Grant bdr.replication_status to bdr_read_all_stats (BDR-2523) <p>  Alllow the read_all_stats role to check BDR replication information. </p>                                                                                                                                                                             |
+| BDR       | 4.2.1   | Bug fix | Resolve hang related to `bdr.standby_slots_min_confirmed` (BDR-2445, RT80692) <p> Allow the WAl sender to proceed without the need to wait for the timeout after a Subscriber Only node is parted, while using standby_slot_names. </p>                                                                                  |
+| BDR       | 4.2.1   | Bug fix | Use two_phase slots in `bdr_init_physical` on PG14+ () <p> Improve prepared transaction handling during `bdr_init_physical`. </p>                                                                                                                                                                                        |
```
Comment (drothery-edb) [inline product_docs/docs/pgd/4/rel_notes/pgd_4.2.1_rel_notes.mdx:12]:
```suggestion
| BDR       | 4.2.1   | Bug fix | Use two_phase slots in `bdr_init_physical` on PG14+ () <p> Improve prepared transaction handling during `bdr_init_physical`. </p>                                                                                                                                                                                        |
```
Is there a bug number for this? If not, let's remove the parens.

Comment (drothery-edb) [reply product_docs/docs/pgd/4/rel_notes/pgd_4.2.1_rel_notes.mdx:12]: updated the landing page for release notes. Please review and update the date and versions, if needed

File: product_docs/docs/pgd/4/rel_notes/pgd_4.2.1_rel_notes.mdx [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+| BDR       | 4.2.1   | Bug fix | Grant `bdr.replication_status to `bdr_read_all_stats` (BDR-2523) <p>  Alllow the `read_all_stats` role to check BDR replication information. </p>                                                                                                                                                                             |
+| BDR       | 4.2.1   | Bug fix | Resolve hang related to `bdr.standby_slots_min_confirmed` (BDR-2445, RT80692) <p> Allow the WAl sender to proceed without the need to wait for the timeout after a Subscriber Only node is parted, while using `standby_slot_names`. </p>                                                                                  |
+| BDR       | 4.2.1   | Bug fix | Use two_phase slots in `bdr_init_physical` on PG14+<p> Improve prepared transaction handling during `bdr_init_physical`. </p>                                                                                                                                                                                        |
```
Comment (irionr) [inline product_docs/docs/pgd/4/rel_notes/pgd_4.2.1_rel_notes.mdx:12]: Is the space needed after `PG14+` ?

File: product_docs/docs/pgd/3.6/release-notes.mdx [hunk 5]
```diff
@@ -1,7 +1,23 @@
...
+- Fix bdr.alter_table_conflict_detection() (BDR-2650)  
+  Correct the function to use the 'detector_args' argument.
+
+### Upgrades
+
+This release supports upgrading from the following versions of BDR:
+
+ - 3.6.20 and higher
+
 ## BDR 3.6.2 (2022 Aug 23)
```
Comment (irionr) [inline product_docs/docs/pgd/3.6/release-notes.mdx:21]: I see there is an issue with the previous release, it should be `3.6.32`, not `3.6.2`

**Review by drothery-edb — APPROVED** [review]

**Pull request merged**